addons/stock/migrations/13.0.1.1/post-migration.py

```diff
@@ -0,0 +1,322 @@
+# Copyright 2020 Payam Yasaie <https://www.tashilgostar.com>
+# Copyright 2020 ForgeFlow <https://www.forgeflow.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+_unlink_by_xmlid = [
+    # ir.property
+    'stock.property_stock_inventory',
+    'stock.property_stock_production',
+    # ir.sequence
+    'stock.sequence_stock_scrap',
+]
+
+
+def fill_company_id(cr):
+    # stock.move.line
+    openupgrade.logged_query(
+        cr, """
+        UPDATE stock_move_line sml
+        SET company_id = sm.company_id
+        FROM stock_move sm
+        WHERE sml.move_id = sm.id AND sml.company_id IS NULL"""
+    )
+    # stock.picking.type
+    openupgrade.logged_query(
+        cr, """
+        UPDATE stock_picking_type spt
+        SET company_id = sw.company_id
+        FROM stock_warehouse sw
+        WHERE spt.warehouse_id = sw.id AND spt.company_id IS NULL"""
+    )
+    # stock.picking
+    openupgrade.logged_query(
+        cr, """
+        UPDATE stock_picking sp
+        SET company_id = spt.company_id
+        FROM stock_picking_type spt
+        WHERE sp.picking_type_id = spt.id AND sp.company_id IS NULL"""
+    )
+    # stock.package_level
+    openupgrade.logged_query(
+        cr, """
+        UPDATE stock_package_level spl
+        SET company_id = sqp.company_id
+        FROM stock_quant_package sqp
+        WHERE spl.package_id = sqp.id AND spl.company_id IS NULL"""
+    )
+    # stock.scrap
+    openupgrade.logged_query(
+        cr, """
+        UPDATE stock_scrap ss
+        SET company_id = ru.company_id
+        FROM res_users ru
+        WHERE ru.id = ss.create_uid AND ss.company_id IS NULL"""
+    )
+    # stock.putaway.rule
+    openupgrade.logged_query(
+        cr, """
+        UPDATE stock_putaway_rule spr
+        SET company_id = ru.company_id
+        FROM res_users ru
+        WHERE ru.id = spr.create_uid AND spr.company_id IS NULL"""
+    )
+    # stock.production.lot
+    openupgrade.logged_query(
+        cr, """
```

Comment:
I really don't like this from a multicompany perspective. I think it needs more thought. Something more like

`SELECT company_id from stock_move_line sml WHERE sml.lot_id = spl.id and sml.company_id IS NOT NULL LIMIT 1` 

but even then you are will have cases where stock has transferred between companies, so you are going to need to do more advanced things like split the lot in two and rewrite some records.

Reply:
not sure, neither agree nor disagree